packages/dev/core/src/Meshes/Compression/dracoEncoder.ts

```diff
@@ -0,0 +1,293 @@
+type DracoEncoderModule = (props: { wasmBinary?: ArrayBuffer }) => Promise<EncoderModule>;
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+declare let DracoEncoderModule: DracoEncoderModule;
+
+/**
+ * Map the Babylon.js attribute kind to the Draco attribute kind, defined by the `GeometryAttributeType` enum.
+ * @internal
+ */
+function getDracoAttributeName(kind: string): DracoAttributeName {
+    if (kind === VertexBuffer.PositionKind) {
+        return "POSITION";
+    } else if (kind === VertexBuffer.NormalKind) {
+        return "NORMAL";
+    } else if (kind === VertexBuffer.ColorKind) {
+        return "COLOR";
+    } else if (kind.startsWith(VertexBuffer.UVKind)) {
+        return "TEX_COORD";
+    }
+    return "GENERIC";
+}
+
+/**
+ * Get the indices for the geometry, if present. Eventually used as
+ * `AddFacesToMesh(mesh: Mesh, numFaces: number, faces: Uint16Array | Uint32Array)`;
+ * where `numFaces = indices.length / 3` and `faces = indices`.
+ * @internal
+ */
+function PrepareIndicesForDraco(input: Mesh | Geometry): Nullable<Uint32Array | Uint16Array> {
+    // Process indices
+    let indices = input.getIndices();
+
+    // Convert number[] to typed array, if needed
+    if (indices && !(indices instanceof Uint32Array) && !(indices instanceof Uint16Array)) {
+        indices = (indices.length > 65535 ? Uint32Array : Uint16Array).from(indices);
+    }
+
+    return indices;
+}
+
+/**
+ * Get relevant information about the geometry's vertex attributes for Draco encoding. Eventually used for each attribute as
+ * `AddFloatAttribute(mesh: Mesh, attribute: number, count: number, itemSize: number, array: TypedArray)`
+ * where `attribute = EncoderModule[<dracoAttribute>]`, `itemSize = <size>`, `array = <data>`, and count is the number of position vertices.
+ * @internal
+ */
+function PrepareAttributesForDraco(input: Mesh | Geometry, excludedAttributes?: string[]): Array<IDracoAttributeData> {
+    const attributes: Array<IDracoAttributeData> = [];
+
+    for (const kind of input.getVerticesDataKinds()) {
+        if (excludedAttributes?.includes(kind)) {
+            if (kind === VertexBuffer.PositionKind) {
+                throw new Error("Cannot exclude position attribute from Draco encoding.");
+            }
+            continue;
+        }
+
+        // Convert number[] to typed array, if needed
+        let data = input.getVerticesData(kind)!;
+        if (!(data instanceof Float32Array)) {
+            data = Float32Array.from(data!);
+        }
+        attributes.push({ babylonAttribute: kind, dracoAttribute: getDracoAttributeName(kind), size: input.getVertexBuffer(kind)!.getSize(), data: data });
+    }
+
+    return attributes;
+}
+
+/**
+ * Copies `data` of each `attribute` into a single buffer for fast transfer to the worker.
+ * @internal
+ */
+function UseTransferableBuffer(attributes: Array<IDracoAttributeData>, indices: Nullable<Uint32Array | Uint16Array>): ArrayBuffer {

```

Comment:
`indices` are not being updated, so the original is being copied again. We talked offline and maybe we just use `slice` with multiple transferrable objects instead.